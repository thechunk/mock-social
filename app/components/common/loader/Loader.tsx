import * as React from 'react';
import {PureComponent} from "react";
import {ActivityIndicator, ActivityIndicatorProps, View} from "react-native";
import styles from './styles';

export default class Loader extends PureComponent<ILoader & ActivityIndicatorProps> {
    render() {
        return this.props.loading ? (
            <View style={[styles.loader, this.props.style]}>
                <ActivityIndicator size="small" {...this.props as ActivityIndicatorProps} />
            </View>
        ) : null;
    }
}