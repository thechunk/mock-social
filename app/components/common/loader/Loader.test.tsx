import * as React from 'react';
import {shallow} from "enzyme";
import Loader from "./Loader";

describe('Loader', () => {
    it('renders correctly', () => {
        expect(shallow(<Loader loading={true}/>)).toMatchSnapshot();
    });

    it('renders null if not loading', () => {
        expect(shallow(<Loader loading={false}/>)).toMatchSnapshot();
    });
});