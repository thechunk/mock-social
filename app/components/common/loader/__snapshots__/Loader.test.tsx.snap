// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    loading={true}
    size="small"
  />
</View>
`;

exports[`Loader renders null if not loading 1`] = `""`;
